samples/dotnet/FileCompression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.FileCompression;
+
+//**********************************************************************************************************************
+// EXAMPLE USAGE
+//
+// ISemanticKernel kernel = SemanticKernel.Build();
+// var zipCompressor = new ZipFileCompressor();
+// var skill = new FileCompressionSkill(zipCompressor);
+// var fileCompression = kernel.ImportSkill(skill, "FileCompression");
+// string sourceFilePath = "FileToCompress.txt";
+// string destinationFilePath = "CompressedFile.zip";
+// var variables = new ContextVariables(sourceFilePath);
+// variables.Set(FileCompressionSkill.Parameters.DestinationFilePath, destinationFilePath);
+// await kernel.RunAsync(variables, fileCompression["CompressFileAsync"]);
+//**********************************************************************************************************************
+
+/// <summary>
+/// Skill for compressing and decompressing files.
+/// </summary>
+public class FileCompressionSkill
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Directory to which to extract compressed file's data.
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// File path where to save compressed data.
+        /// </summary>
+        public const string DestinationFilePath = "destinationFilePath";
+    }
+
+    private readonly IFileCompressor _fileCompressor;
```

Comment:
Current style guidance is to order all private members below public members.

Reply:
Changed to go with current guidance.  In the future, we might want to consider one where we have the following:

All public properties
All protected properties
All private properties

All public methods
All protected methods
All private methods

And so on...

I believe that's the default in StyleCop.

Either way, we can enforce what we choose through StyleCop.

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md